### DISCUSSION TITLE
How to disable debug prints in mosaik 3.2?

### DISCUSSION THREAD

**Discussion by valerianoCarlos** [OFFIS-mosaik/mosaik#17] [Q&A] [answered]
Hi everyone, I am currently using the Mosaik framework for a personal project involving several Python simulators. I noticed that in the latest versions of Mosaik (3.2 onwards), the console shows DEBUG prints for all simulator functions being called at each step, making it very difficult to navigate through the output. Here’s a couple of examples of what I’m referring to:

`2024-07-15 17:22:04.068 | DEBUG    | mosaik_api_v3:run:530 - Calling setup_done(*[], **{})`
`2024-07-15 17:22:04.069 | DEBUG    | mosaik_api_v3:run:530 - Calling step(...)`
`2024-07-15 17:22:04.073 | DEBUG    | mosaik_api_v3:run:530 - Calling get_data(...)`

Is there any way to disable or suppress these debug prints? Your help would be greatly appreciated.

**Comment by eike** [comment]
Yeah, this is an unfortunate side effect of switching over to Loguru for the mosaik API package, which we noticed too late. For now, you can disable this output by explicitly setting the `LOGURU_LEVEL` environment variable for remote simulators, like so:
```python
SIM_CONFIG = {
  "SimlutatorName": {
    "cmd": "<your command as before">,
    "env": {"LOGURU_LEVEL": "INFO"},
  },
}
```

I have also created an issue to remove these outputs by default ([#242](https://gitlab.com/mosaik/mosaik/-/issues/242)). Now that I think about it, the real solution should also be very simple; if it is, I might be able to do this today.

**Comment by eike** [comment, marked as answer]
It took an additional day, but we just published a new minor version of the mosaik-api-v3 package (namely 3.0.11). When you update to that, the DEBUG messages should be gone. This API package should work fine with both mosaik 3.2 and mosaik 3.3.

**Comment by valerianoCarlos** [comment]
Thank you for your help. I noticed that version 3.0.11 is not yet available on PyPI, so I'm unsure how to update the package to that version without using pip. Once it is released, I will definitely check it out. Thank you again for your support!

**Reply by eike** [reply]
Sorry, I forgot the last button click in our pipeline. It's on PyPI now.
